lib/connection/result/row_stream.js

```diff
@@ -173,7 +173,40 @@ function RowStream(statement, context, options)
     // add the next row to the read queue
     process.nextTick(function ()
     {
-      self.push(externalizeRow(row, columns, mapColumnIdToExtractFnName));
+      // check if there are still rows available in the rowBuffer
+      if (rowBuffer)
+      {
+        rowIndex--; // decrement the index to include the previous row in the while loop
```

Comment:
are we sure that `rowIndex` is not negative after decrement? some lines above it's set to `0`

Reply:
The line before it always increments the rowIndex by 1: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/505/files/7e6cb500a769f6bb561440f9cd327dcc0c2e4e2c#diff-2e25c6e02363450501da256765b53482373c3976dfdc0dee01bd1c1160666c87R151

It's possible to do some refactoring so the decrement is no longer needed

Reply:
@sfc-gh-ext-simba-lf in one condition it set to 0 also in above code.

Reply:
In the condition that rowIndex is set to 0, the rowBuffer is also set to null: https://github.com/snowflakedb/snowflake-connector-nodejs/pull/505/files/7e6cb500a769f6bb561440f9cd327dcc0c2e4e2c#diff-2e25c6e02363450501da256765b53482373c3976dfdc0dee01bd1c1160666c87L158

If the rowBuffer is null, the decrement does not occur since it goes to the [else statement](https://github.com/snowflakedb/snowflake-connector-nodejs/pull/505/files/7e6cb500a769f6bb561440f9cd327dcc0c2e4e2c#diff-2e25c6e02363450501da256765b53482373c3976dfdc0dee01bd1c1160666c87R205)

If the [rowBuffer is not null](https://github.com/snowflakedb/snowflake-connector-nodejs/pull/505/files/7e6cb500a769f6bb561440f9cd327dcc0c2e4e2c#diff-2e25c6e02363450501da256765b53482373c3976dfdc0dee01bd1c1160666c87R177) then the rowIndex was not set to 0, and the decrement happens

Reply:
Would you like for this to be refactored to remove the decrement?

Reply:
It would be nice to have this decrement refactored or in `if` above we could verify if it's `>0` next to `rowBuffer` existance check

Reply:
Ok, added a check to the if statement to make sure rowIndex is > 0